src/stringProcessing/matchTextWithArray.js

```diff
@@ -18,6 +19,9 @@ import { uniq as unique } from "lodash-es";
 export default function( text, array, locale = "en_EN" ) {
 	let count = 0;
 	let matches = [];
+
+	array = array.map( normalizeQuotes );
```

Comment:
`matchTextWithArray` makes use of `matchTextWithWord` (line 26), which [already normalizes quotes](https://github.com/Yoast/YoastSEO.js/blob/develop/src/stringProcessing/matchTextWithWord.js#L25). Could you check if this normalization here is actually needed?

Reply:
It is needed because the morphology researcher automatically expands keywords with quotes to include other types of quotes:
```"key'word" => ["key`word", "key'word", "key‘word", (...) ]```

`matchTextWithArray` loops over all keyword forms (three in the case above) and tries to match them in the string.
Since `matchTextWithWord` normalizes the quotation marks, each expanded keyword matches on all keywords in the text.

By normalizing the quotes again beforehand, the application of `unique` on the array (line 25) filters out the expanded quotes again.

This is a quick and dirty fix. I think we can solve this in a cleaner way by either:
* Do not expand quotes during the building of all keyword forms.
* Do not normalize quotes during matching.

Reply:
Ah yeah, I see the problem. The only downside now is that we push the normalized version into the `matches` array. This is later used for markings. Since we push the normalized `key'word` for instances of `key`word`, `key`word` won't be marked in the text. But I think for now that's okay. As you say we need to find a more standardized approach to deal with markers (HTML parser!).